src/index.js

```diff
@@ -481,6 +493,13 @@ export default class RNPickerSelect extends PureComponent {
                     supportedOrientations={['portrait', 'landscape']}
                     onOrientationChange={this.onOrientationChange}
                     {...modalProps}
+                    onShow={() => {
+                        if (this.context) {
+                            this.context.setIsModalShown(true);
+                        }
```

Comment:
If we show the Modal, `onShow` callback will be called and set modal as shown (`isModalShown=true`). Now if we unmount that modal (while it's shown) the state of the modal being shown won't be reset accordingly - state is stuck at `true`. We should have cleared the state on `componentWillUnmount`. (Coming from https://github.com/Expensify/App/issues/23044)